### PR TITLE
make jwt cookie available for subdomains

### PR DESCRIPTION
**Pull request by tofixx**
the cookie domain can be defined using env SC_DOMAIN, if given, the cookie will be available for subdomains as well. if not defined the domain parameter will not be set in cookie header definition

### CODE REVIEW

**Comment by MrSerth** [comment]
Do we know what added the `www.` accidentally in our production deployment?

**Comment by tofixx** [comment]
I would recommend to Change the value to be from configuration.

**Comment by MrSerth** [comment]
Alright, if there is no other (dynamic) way, then we should do that. Otherwise, this PR is the same as #981.

File: controllers/registration.js [hunk 1]
```diff
@@ -65,7 +65,7 @@ ${res.locals.theme.short_title}-Team`
         });
     }).then(function() {
         if (req.params.sso) {
-            res.cookie('jwt', req.cookies.jwt, {expires: new Date(Date.now() - 100000)});
+            res.cookie('jwt', req.cookies.jwt, { domain: req.hostname, expires: new Date(Date.now() - 100000) });
```
Comment (tofixx) [inline controllers/registration.js]: we should update this to clearCookie instead

File: controllers/login.js [hunk 2]
```diff
@@ -37,7 +37,7 @@ router.post('/login/', function (req, res, next) {
             } else {
                 const login = (data) => {
                     return api(req).post('/authentication', {json: data}).then(data => {
-                        res.cookie('jwt', data.accessToken, {expires: new Date(Date.now() + 30 * 24 * 60 * 60 * 1000)});
+                        res.cookie('jwt', data.accessToken, { domain: req.hostname, expires: new Date(Date.now() + 30 * 24 * 60 * 60 * 1000) });
```
Comment (tofixx) [inline controllers/login.js]: alternative strategy: use config value or ignore domain property

**Comment by adrianjost** [comment]
@tofixx please resolve conflicts

**Comment by tofixx** [comment]
Dank vieler Merges bereits im Master, getestet in FF und Chrome.

**Pull request closed without merging**